server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -46,169 +44,191 @@
         return creationMessage.getFrom()
-            .map(draftEmailer -> draftEmailer
-                .getEmail()
-                .map(mail -> validationResultForIncorrectAddress(allowedSenders, mail))
-                .orElse(Lists.newArrayList()))
-            .orElse(Lists.newArrayList());
+                .map(draftEmailer -> draftEmailer.getEmail()
+                        .map(allowedFromMailAddresses::contains)
+                        .orElse(false))
+                .orElse(false);
     }
 
-    private List<ValidationResult> validationResultForIncorrectAddress(List<String> allowedSenders, String mail) {
-        if (!allowedSenders.contains(mail)) {
-            return Lists.newArrayList(ValidationResult.builder()
-                .message("Invalid 'from' field. Must be one of " + allowedSenders)
-                .property(MessageProperty.from.asFieldName())
-                .build());
+    
+    private MessageWithId handleOutboxMessages(CreationMessageEntry entry, MailboxSession session) throws MailboxException, MessagingException {
+        Mailbox outbox = getMailboxWithRole(session, Role.OUTBOX).orElseThrow(() -> new MailboxNotFoundException(Role.OUTBOX.serialize()));
+        if (!isRequestForSending(entry.getValue(), session)) {
+            throw new MailboxUnhandledRequestException();
         }
-        return Lists.newArrayList();
+        Function<Long, MessageId> idGenerator = uid -> generateMessageId(session, outbox, uid);
+        return createMessageInOutboxAndSend(entry, session, outbox, idGenerator);
     }
-
-    private Map<CreationMessageId, SetError> handleCreationErrors(Predicate<CreationMessage> invalidMessagesTester,
-                                                                  Function<CreationMessage, List<ValidationResult>> toValidationResults,
-                                                                  SetMessagesRequest request) {
-        return request.getCreate().entrySet().stream()
-                .filter(e -> invalidMessagesTester.test(e.getValue()))
-                .map(e -> new AbstractMap.SimpleEntry<>(e.getKey(), buildSetErrorFromValidationResult(toValidationResults.apply(e.getValue()))))
-                .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+    
+    @VisibleForTesting
+    protected MessageWithId createMessageInOutboxAndSend(CreationMessageEntry createdEntry,
+                                                           MailboxSession session,
+                                                           Mailbox outbox, Function<Long, MessageId> buildMessageIdFromUid) throws MailboxException, MessagingException {
+        CreationMessageId creationId = createdEntry.getCreationId();
+        MessageMapper messageMapper = mailboxSessionMapperFactory.createMessageMapper(session);
+        MailboxMessage newMailboxMessage = buildMailboxMessage(createdEntry, outbox);
+        messageMapper.add(outbox, newMailboxMessage);
```

Comment:
That is the way.MessageResult
MailboxMessage is a "store" concept, it is not usable at the "api" level. 

Then what you need is something to change your MessageResult into your jmap model. I would see no troubbles adding the fields you need from mailboxMessage to the MessageResult, and thus making the "api" richer.

(It will be needed anyway when not using mappers in jmap for message operations)

Globally by making some information available at the store level you can avoid several DB roundtrips (See MailboxUtil). Moving this information upper also makes sens in my opinion. Not every protocol needs these information, but we should retrieve it with the API for those which do.